internal/runtime/storeforward.go

```diff
@@ -52,33 +71,45 @@ func (sf *storeForwardInfo) startStoreAndForwardRetryLoop(
 	enabledWg.Add(1)
 
 	config := container.ConfigurationFrom(sf.dic.Get)
-	lc := bootstrapContainer.LoggingClientFrom(sf.dic.Get)
+	storeClient := container.StoreClientFrom(sf.dic.Get)
+
+	sf.serviceKey = serviceKey
+
+	items, err := storeClient.RetrieveFromStore(serviceKey)
+	if err != nil {
+		sf.lc.Errorf("Unable to initialize Store and Forward data count: Failed to load items from DB: %v", err)
+	} else {
+		sf.dataCount.Clear()
+		sf.dataCount.Inc(int64(len(items)))
+	}
 
 	go func() {
 		defer appWg.Done()
 		defer enabledWg.Done()
 
 		retryInterval, err := time.ParseDuration(config.Writable.StoreAndForward.RetryInterval)
 		if err != nil {
-			lc.Warn(
+			sf.lc.Warn(
 				fmt.Sprintf("StoreAndForward RetryInterval failed to parse, defaulting to %s",
 					defaultMinRetryInterval.String()))
 			retryInterval = defaultMinRetryInterval
 		} else if retryInterval < defaultMinRetryInterval {
-			lc.Warn(
+			sf.lc.Warn(
 				fmt.Sprintf("StoreAndForward RetryInterval value %s is less than the allowed minimum value, defaulting to %s",
 					retryInterval.String(), defaultMinRetryInterval.String()))
 			retryInterval = defaultMinRetryInterval
 		}
 
 		if config.Writable.StoreAndForward.MaxRetryCount < 0 {
-			lc.Warn("StoreAndForward MaxRetryCount can not be less than 0, defaulting to 1")
+			sf.lc.Warn("StoreAndForward MaxRetryCount can not be less than 0, defaulting to 1")
```

Comment:
what's the difference between `sf.lc` and `appContext.LoggingClient()` used here vs line 151 respectively?

Reply:
refactored